Detections/CommonSecurityLog/TimeSeriesAnomaly-MultiVendor_NetworkTraffic.yaml

```diff
@@ -51,15 +55,21 @@ query: |
   let TimeSeriesAlerts=TimeSeriesData
   | extend (anomalies, score, baseline) = series_decompose_anomalies(Total, scorethreshold, -1, 'linefit')
   | mv-expand Total to typeof(double), TimeGenerated to typeof(datetime), anomalies to typeof(double),score to typeof(double), baseline to typeof(long)
-  | where anomalies > 0 | extend score = round(score,2)
-  | project DeviceVendor,TimeGenerated, Total, baseline, anomalies, score;
+  | where anomalies > 0 | extend score = round(score,2), AnomalyHour = TimeGenerated
+  | project DeviceVendor,AnomalyHour, TimeGenerated, Total, baseline, anomalies, score;
   //Join anomalies with Base Data to popalate associated records for investigation - Results sorted by score in descending order 
   TimeSeriesAlerts 
   | join (
       CommonSecurityLog 
   | where isnotempty(DestinationIP) and isnotempty(SourceIP)
   | where TimeGenerated between (startofday(ago(starttime))..startofday(ago(endtime)))
-  | summarize HourlyCount = count(), ListOfDestPorts = make_list(DestinationPort)  by SourceIP, DestinationIP,DestinationPort, DeviceVendor, bin(TimeGenerated, 1h)
-  ) on TimeGenerated, DeviceVendor | extend PercentTotal = round((HourlyCount / Total) * 100, 3) | sort by score desc 
-  | project DeviceVendor ,TimeGenerated, SourceIP, DestinationIP, ListOfDestPorts, HourlyCount, PercentTotal, Total, baseline, score, anomalies
-  | extend IPCustomEntity = DestinationIP
+  | summarize HourlyCount = count(), TimeGeneratedMax = arg_max(TimeGenerated, *), DestinationIPlist = make_set(DestinationIP), DestinationPortlist = make_set(DestinationPort) by DeviceVendor, SourceIP, TimeGeneratedHour= bin(TimeGenerated, 1h)
+  | extend AnomalyHour = TimeGeneratedHour
+  ) on AnomalyHour, DeviceVendor 
+  | extend PercentTotal = round((HourlyCount / Total) * 100, 3)
+  | where PercentTotal > percentotalthreshold
+  | project DeviceVendor , AnomalyHour, TimeGeneratedMax, SourceIP, DestinationIPlist, DestinationPortlist, HourlyCount, PercentTotal, Total, baseline, score, anomalies
```

Comment:
Same here, any reason to not go right to the summarize instead of doing project first?

Reply:
I noticed for CommonSecurityLog , adding project before summarize improved my query time drastically. If I don`t have project before summarize, query time outs, but after adding project it completes without issues. Seems by limiting schema of output table improving performance in some cases.